### DISCUSSION TITLE
playing on a specific core - and - playing from head file

### DISCUSSION THREAD

**Discussion by sartyx** [schreibfaul1/ESP32-audioI2S#690] [General]
As this lib is especially for S3 I wonder if it is possible to let it run on a specific core.
Also what has been asked before from internal memory / head file such as

```
unsigned const char WavData[35564] = {
  0x52, 0x49, 0x46, 0x46, 0xe4, 0x8a, 0x00, 0x00, 0x57, 0x41, 0x56, 0x45,
  0x66, 0x6d, 0x74, 0x20, 0x10, 0x00, 0x00, 0x00, 0x01, 0x00, 0x01, 0x00, .....
```

**Comment by schreibfaul1** [comment]
Reading from an array is not intended. To do this, you need a routine that reads the array and writes it to the audio buffer, just as SD.read() does for files. An alternative would be to create an FFat partition and store the array in it as a file.

**Comment by sartyx** [comment]
I guess I need to find out how to use the spiffs/ffat then. The arduino IDE simply responds with SPIFFS not supported on ESP32.
